### ISSUE TITLE
ImageSlideShow shows blank screen when I change the input source list to a single item list

### ISSUE THREAD

**Issue by soleilpqd** [zvonicek/ImageSlideshow#338]
Hi,

I have an old project in Swift 3 with ImageSlideShow 1.3.2 (XCode 8). It works fine like this:
- I show a dialog of ImageSlideShow with a list of some InputSource items (name as list A).
- When user taps Delete button on dialog, I delete the current InputSource item in list A, then I call ImageSliceShow.setImageInputs(A) to reload the slide show.

Now I upgrade the project up to Swift 5 with ImageSlideShow latest (1.8.0) (with XCode 10.2 - latest iOS SDK). There's an issue:
- It works fine if I show ImageSlideShow dialog with list A containing 1 item (I mean setImageInputs(A) works fine at first time).
- With list A count > 2, when I delete item in A and call setImageInputs(A) to reload the slide show, it works fine.
- But when list A has 2 items, I delete 1 item so list A now contains only 1, then I call setImageInputs(A). ImageSlideShow now shows  blank screen. (In Debug View Hierarchy, there's no subview in ImageSlideShow.scrollview).

I apply that deleting spec into example project & get same issue.

Please check, thank you!

**Comment by biqinglin** [comment]
I have the same problem as you...

**Comment by ToMark1881** [comment]
same problem

**Comment by aornano** [comment]
Sure same problem in you try to load with placeholder setted an errored image like: http://upload.wikimedia.org/wikipedia/it/4/4b/ConchettaNaviglioPavese.jpg

Server says: File not found: /v1/AUTH_mw/wikipedia-it-local-public.4b/4/4b/ConchettaNaviglioPavese.jpg but ImageSlideShow showed a blank image..

**Comment by ApolloZhu** [comment]
Before this gets fixed, you could clear all the inputs first, then reload ImageSlideshow with actual inputs as a temporary workaround:

```swift
slideshow.setImageInputs([])
slideshow.setImageInputs(actualInputs)
```

**Comment by tkhelm** [comment]
Same problem...following.

**Comment by kunass2** [comment]
Same problem;)

**Comment by kunass2** [comment]
> Before this gets fixed, you could clear all the inputs first, then reload ImageSlideshow with actual inputs as a temporary workaround:
> 
> ```swift
> slideshow.setImageInputs([])
> slideshow.setImageInputs(actualInputs)
> ```

Worked perfectly;)

**Comment by Dibyaa** [comment]
@ApolloZhu  I am getting EXC_BAD_ACCESS while building. Can you help me out please..
I have attached the screenshot below:

![Screen Shot 2019-08-26 at 5 36 00 PM](https://user-images.githubusercontent.com/48793541/63853302-0e264580-c9ba-11e9-8e7a-604a54dc6e9c.png)

**Comment by zvonicek** [comment]
This should be fixed in version 1.8.1.

**Comment by jizambrana** [comment]
Hi,

I have a similar issue with version 1.8.1 and Kingfisher version 5.8.1. 
Every time I reload the ImageSlideShow with a new array using .setImageInputs() (because sometimes I need to add new images to the array), it only shows the newest images. Old images are shown blank.
Perhaps there is another way to add new images that I don't know, but even if I do .setImageInputs([]) before calling it with the new images doesn't work.

If anyone could help me, thank you!!
